src/pages/index.js

```diff
@@ -143,29 +237,69 @@ function getCardElement(data) {
   const cardNameEl = cardElement.querySelector(".card__title");
   const cardLikeBtn = cardElement.querySelector(".card__like-btn");
   const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+  cardElement.dataset.id = data._id;
   cardNameEl.textContent = data.name;
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_active");
+  } else {
+    cardLikeBtn.classList.remove("card__like-btn_active");
+  }
   cardLikeBtn.addEventListener("click", () => {
-    cardLikeBtn.classList.toggle("card__like-btn_liked");
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_active");
+
+    api
+      .toggleLike(data._id, isLiked)
+      .then((updatedCard) => {
+        cardLikeBtn.classList.toggle(
+          "card__like-btn_active",
+          updatedCard.isLiked
+        );
+      })
+      .catch((err) => console.error(err));
   });
+
   cardImageEl.addEventListener("click", () => {
     previewModalCaptionEl.textContent = data.name;
     previewModalImageEl.src = data.link;
     previewModalImageEl.alt = data.name;
+    console.log("Image clicked. Opening preview modal.");
     openModal(previewModal);
   });
+
   cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, data);
   });
+
   return cardElement;
 }
 
-// Populate Initial Cards
-initialCards.forEach((item) => {
-  const cardEl = getCardElement(item);
-  cardsList.append(cardEl);
-});
+let selectedCard = null;
+let selectedCardId = null;
+
+// Delete modal
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const deleteButton = deleteForm.querySelector(".modal__submit-btn_delete");
```

Comment:
`  const deleteButton = deleteForm.querySelector(".modal__submit-btn_delete");
`

Please, find this button only once outside the function

Reply:
`  const deleteButton = deleteForm.querySelector(".modal__submit-btn_delete");
`

Please, find this button only once outside the function

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine